app/src/main/java/com/mdg/droiders/samagra/shush/adapters/TimeListAdapter.java

```diff
@@ -0,0 +1,495 @@
+package com.mdg.droiders.samagra.shush.adapters;
+
+import android.content.ContentValues;
+import android.content.Context;
+import android.database.Cursor;
+import android.net.Uri;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.CheckBox;
+import android.widget.CompoundButton;
+import android.widget.Switch;
+import android.widget.TextView;
+
+import com.mdg.droiders.samagra.shush.AlarmScheduler;
+import com.mdg.droiders.samagra.shush.R;
+import com.mdg.droiders.samagra.shush.data.PlacesContract;
+import com.mdg.droiders.samagra.shush.fragments.TimePickerDialogFragment;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Locale;
+
+/**
+ * Created by rohan on 19/8/17.
+ * Adapter that binds data to the view for time entries in the Shush app.
+ */
+public class TimeListAdapter extends RecyclerView.Adapter<TimeListAdapter.TimeRowHolder> {
+
+    private static final String DIALOG_TIME_TAG = "time_picker_dialog_tag";
+    // The format which is used to display date in each row
+    private static final SimpleDateFormat DISPLAY_DATE_FORMAT =
+            new SimpleDateFormat("hh:mm a", Locale.ENGLISH);
+    private static final String LOG_TAG = "Samagra/TLA/";
+
+    private Context mContext;
+    private Cursor timeDataCursor;
+    private AlarmScheduler alarmScheduler;
+
+    public TimeListAdapter(Context mContext) {
+        this.mContext = mContext;
+        refreshCursor();
+        alarmScheduler = new AlarmScheduler(mContext);
+    }
+
+    @Override
+    public TimeRowHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View row = LayoutInflater.from(mContext)
+                .inflate(R.layout.time_picker_row, parent, false);
+        return new TimeRowHolder(row);
+    }
+
+    @Override
+    public void onBindViewHolder(TimeRowHolder holder, int position) {
+        if (position == timeDataCursor.getCount()) {
+            holder.itemView.setVisibility(View.INVISIBLE);
+            return;
+        }
+        // isBound set to false so that listeners do not react to set events in this method.
+        holder.isBound = false;
+        holder.itemView.setVisibility(View.VISIBLE);
+        if (timeDataCursor.moveToPosition(position)) {
+
+            // Retrieving the whole row from the database.
+            int id = timeDataCursor.getInt(
+                    timeDataCursor.getColumnIndexOrThrow(PlacesContract.TimeEntry._ID)
```

Comment:
No, but since we are using the column index from the contract class, we can be sure that the method call will not result in an unhandled exception state.